From 1200fce02e1c8a9ae41efc769e2075ab5ae0d776 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Tue, 14 Oct 2014 00:38:28 -0500
Subject: [PATCH 01/13] make sure that the filesystem is a type of fat

parted marks the partition table as being fat32 though we may pyt a fat16
filesystem on it.
---
 fs/fat/fat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/fat/fat.c b/fs/fat/fat.c
index 561921f..13a8acc 100644
--- a/fs/fat/fat.c
+++ b/fs/fat/fat.c
@@ -74,7 +74,7 @@ int fat_set_blk_dev(block_dev_desc_t *dev_desc, disk_partition_t *info)
 	/* Check for FAT12/FAT16/FAT32 filesystem */
 	if (!memcmp(buffer + DOS_FS_TYPE_OFFSET, "FAT", 3))
 		return 0;
-	if (!memcmp(buffer + DOS_FS32_TYPE_OFFSET, "FAT32", 5))
+	if (!memcmp(buffer + DOS_FS32_TYPE_OFFSET, "FAT", 3))
 		return 0;
 
 	cur_dev = NULL;
-- 
2.1.0

